Framework/Geometry/src/Instrument/InstrumentVisitor.cpp

```diff
@@ -45,7 +45,7 @@ void clearLegacyParameters(ParameterMap *pmap, const IComponent &comp) {
   pmap->clearParametersByName(ParameterMap::rotx(), &comp);
   pmap->clearParametersByName(ParameterMap::roty(), &comp);
   pmap->clearParametersByName(ParameterMap::rotz(), &comp);
-  pmap->clearParametersByName(ParameterMap::scale(), &comp);
+  pmap->clearParametersByName(ParameterMap::isHidden(), &comp);
```

Comment:
Good. So you are clearing the `ParameterMap`. Not sure why you've made additions for checking things in the parameter map as part of `CompareWorkspace` seems like the features are in conflict?